client/src/telethon/_impl/mtsender/sender.py

```diff
@@ -230,54 +232,68 @@ def _enqueue_body(self, body: bytes) -> Future[bytes]:
 
     async def _step_until_receive(self, rx: Future[bytes]) -> bytes:
         while True:
-            await self.step()
+            await self._do_step()
             if rx.done():
                 return rx.result()
 
-    async def step(self) -> list[Updates]:
-        async with self.lock:
-            return await self._step()
+    async def get_updates(self) -> list[Updates]:
+        await self.do_step()
+        return (
+            [self._updates.popleft() for _ in range(ulen)]
+            if (ulen := len(self._updates))
+            else []
+        )
+
+    async def do_step(self) -> None:
+        async with self._lock:
+            await self._do_step()
+
+    async def _do_step(self) -> None:
+        await self.step.__anext__()
```

Comment:
I cannot see how this async iterator gets cleaned up, but perhaps I'm missing something.